http.ts

```diff
@@ -4,37 +4,69 @@ import { TextProtoReader } from "./textproto.ts";
 import { STATUS_TEXT } from "./http_status";
 import { assert } from "./util";
 
+interface Deferred {
+  promise: Promise<{}>;
+  resolve: () => void;
+  reject: () => void;
+}
+
+function deferred(): Deferred {
+  let resolve, reject;
+  const promise = new Promise((res, rej) => {
+    resolve = res;
+    reject = rej;
+  });
+  return {
+    promise, resolve, reject,
+  };
+}
+
 export async function* serve(addr: string) {
   const listener = listen("tcp", addr);
-  while (true) {
-    const c = await listener.accept();
-    yield* serveConn(c);
+  let serveDeferred = deferred();
+  let reqQueue: ServerRequest[] = []; // in case multiple promises are ready
+
+  // Continuously read more requests from conn until EOF
+  // Mutually calling with handleReq
+  const readRequestsFromConn = async (conn: Conn) => {
```

Comment:
Can you rename this to `serveConn` (to keep the similar naming as [the Go implementation)](https://github.com/golang/go/blob/6ff8c1db0bb8f0d149acf0187984f29326c10d56/src/net/http/server.go#L1761-L1762)

Is it possible to make these closures into top-level functions? It would be preferable so that they could be tested independently.

Reply:
Renamed and exposed